compiler/rustc_driver/src/lib.rs

```diff
@@ -1142,7 +1150,7 @@ fn extra_compiler_flags() -> Option<(Vec<String>, bool)> {
 pub fn catch_fatal_errors<F: FnOnce() -> R, R>(f: F) -> Result<R, ErrorReported> {
     catch_unwind(panic::AssertUnwindSafe(f)).map_err(|value| {
         if value.is::<rustc_errors::FatalErrorMarker>() {
-            ErrorReported
+            ErrorReported::new_i_know_what_im_doing()
```

Comment:
This should be a downcast, not just `.is::<...>()`, and `FatalErrorMarker` should contain `ErrorReported`.

Reply:
Unfortunately, this is difficult because `FatalError` is defined in `rustc_span`